### PR TITLE
Proof of concept of #76: Adds true extensibility to can-components

### PR DESCRIPTION
**Pull request by imjoshdean**
Implementation of #76, missing any type of docs and some cleanup but:

- Extended components will extend parent's events for it's internal `can-control`
- Implements a `<super/>` tag to utilize the parent's template within the extended component's view

Example: https://gist.github.com/imjoshdean/8f8b3c30d1f5fdcfbc0d93a40f585893

### CODE REVIEW

File: can-component.js [hunk 1]
```diff
@@ -253,12 +269,63 @@ var Component = Construct.extend(
 					options.tags = {};
 				}
 
+				options.parentRenderer = this.parentRenderer;
+				options.parentViewModel = this.viewModel;
+				options.parentTag = this.parentTag;
+
+				options.tags.super = function superHookup(el, superTagData) {
```
Comment (justinbmeyer) [inline can-component.js]: is it possible to make this generic so the same code isn't duplicated for this and `<content>`?  I know the `<slot>` folks will want similar stuff.  Maybe everyone can work together on that @mikemitchel @Sinjhin

Comment (imjoshdean) [inline can-component.js]: Yeah, I totally agree! The code between `content` and `super` is basically identical, and I presume that `slot` will be similar. I'll reach out.

**Comment by imjoshdean** [comment]
This is still open? Feel free to close it if there is no interest in this since I've since moved on from using CanJS.

Much appreciated on the suggested alternative though! ❤️

**Comment by frank-dspeed** [comment]
@justinbmeyer lets close this?

**Comment by mikemitchel** [comment]
as per discussion with @chasenlehara , closing

**Pull request closed without merging**